src/treegen.cpp

```diff
@@ -183,6 +197,17 @@ treegen::error make_ltree(MMVManip &vmanip, v3s16 p0,
 	std::stack <core::matrix4> stack_orientation;
 	std::stack <v3f> stack_position;
 
+	// fill NodeDefs with unique id if requested
+	if (tree_definition.enable_unique_ids) {
+		u8 unique_id = 0;
+		PcgRandom r((((p0.X << 16) + p0.Y) << 16) + p0.Z);
```

Comment:
It seems interesting to use a pseudorandom algorithm as a hash.  Not necessarily wrong, just interesting.
I think that line would be ideal like this:
 u64 flat_pos = ((u16)p0.X << 32) | ((u16)p0.Y << 16) | ((u16)p0.Z << 0);
 PcgRandom r(flat_pos);